experimental/devices/sn3218/sn3218.go

```diff
@@ -0,0 +1,136 @@
+// Copyright 2016 The Periph Authors. All rights reserved.
+// Use of this source code is governed under the Apache License, Version 2.0
+// that can be found in the LICENSE file.
+
+package sn3218
+
+import (
+	"errors"
+
+	"periph.io/x/periph/conn/i2c"
+)
+
+const (
+	i2cAddress             = 0x54
+	cmdEnableOutput        = 0x00
+	cmdSetBrightnessValues = 0x01
+	cmdEnableLeds          = 0x13
+	cmdUpdate              = 0x16
+	cmdReset               = 0x17
+)
+
+type dev struct {
+	i2c        *i2c.Dev
+	states     [18]bool
+	brightness [18]byte
+}
+
+// New returns an object that handles communication with a SN3218 LED driver
+// The SN3213 only accepts writes, so this object also keeps the state of
+// each LED.
+// Important: The device is write only, so it is not possible to read the state
+// of each LED. Therefore it's recommended to run d.Reseet() after this, so that
+// all LEDs are off and have the brightness 0.  Otherwise GetLedState() returns
+// wrong results.
+func New(bus i2c.Bus) (*dev, error) {
+	d := &i2c.Dev{Bus: bus, Addr: i2cAddress}
+	dev := dev{}
+	dev.i2c = d
+	return &dev, nil
+}
+
+// Enable enables the SN3218
+func (d *dev) Enable() error {
+	_, err := d.i2c.Write([]byte{cmdEnableOutput, 0x01})
+	return err
+}
+
+// Disable disables the SN3218
+func (d *dev) Disable() error {
+	_, err := d.i2c.Write([]byte{cmdEnableOutput, 0x00})
+	return err
+}
+
+// Reset resets the registers to the default values
```

Comment:
I don't think you need to expose this function, inline this code inside New().

Reply:
done